test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CustomerRepositoryImpl.kt

```diff
@@ -0,0 +1,67 @@
+package io.micronaut.microstream.docs
+
+import io.micronaut.core.annotation.NonNull
+import jakarta.inject.Singleton
+import one.microstream.concurrency.XThreads
+import one.microstream.storage.embedded.types.EmbeddedStorageManager
+import java.util.*
+import java.util.stream.Collectors
+import javax.validation.Valid
+import javax.validation.constraints.NotBlank
+
+@Singleton
+class CustomerRepositoryImpl(private val embeddedStorageManager: EmbeddedStorageManager) // <1>
+    : CustomerRepository {
+    override fun save(customer: @Valid Customer) {
+        XThreads.executeSynchronized {
+            data.ifPresent { data: Data ->
+                data.add(customer)
+            }
+            embeddedStorageManager.storeAll()
+        }
+    }
+
+    @NonNull
+    override fun findById(id: @NotBlank String): Optional<Customer> {
```

Comment:
Not sure Optional use in kotlin is ideomatic, should be `Customer?` instead?

Reply:
I have removed the `Optional` from Kotlin code: 

https://github.com/micronaut-projects/micronaut-microstream/pull/11/commits/3bbc97ae5d6c92f0008db4d5c1e0efa5c5470423